VWO Cloud Update (#4827)
segmentio/segment-docs: src/connections/destinations/catalog/actions-vwo-cloud/index.md

7. Enable the destination using the toggle switch.

> info ""
> VWO requires you to include a `vwo_uuid` key for all calls made in cloud-mode. The value of this key must be the VWO UUID(in the case of a website) or the User ID (in the case of FullStack). The `vwo_uuid` field must be included in *properties* for Track and Page calls whereas, it should be added in the *traits* for Identify calls.

## Using VWO Cloud mode destination with Web

> info ""
> We recommend using [VWO Web Mode destination](https://segment.com/docs/connections/destinations/catalog/actions-vwo-web/) for webpages as it requires minimal to no additional setup.

1. Install VWO SmartCode on your website. You can follow the steps given [here](https://help.vwo.com/hc/en-us/articles/360019422834-Configuring-SmartCode-for-Your-Website){:target="_blank"} to do the same.
2. Create a VWO campaign on your website.
3. Once a visitor lands on your website a `_vwo_uuid` cookie is generated that acts as a unique identifier for the visitor. To know more about VWO UUID, [click here](https://help.vwo.com/hc/en-us/articles/360034891513-How-to-Locate-your-VWO-UUID-){:target="_blank"}.

> 4. Pass this cookie (*_vwo_uuid*) value with every call to Segment in the `vwo_uuid` key. The `vwo_uuid` key must be included in *properties* for Track and Page calls whereas it should be added in the *traits* for Identify calls.<br><br>To automate this step, you can use [Segment Analytics.js middleware](https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/middleware/){:target="_blank"} and use the following script on your website. This script fetches the VWO UUID from the cookie and adds it to the segment payload so that you don’t have to do the same manually.

Author: This is resolved in a separate commit as there was a minor typo in the suggestion.

      if (parts.length === 2) return parts.pop().split(";").shift();
    };
    const event = payload.obj;
    const vwoUuid = getCookie("_vwo_uuid");
    switch (event.type) {
      case "track":
      case "page":
        payload.obj.properties.vwo_uuid = vwoUuid;
        break;
      case "identify":
        payload.obj.traits.vwo_uuid = vwoUuid;
        break;
    }
    next();
  });
</script>
```

5. All the events triggered in Segment will be available under **UNREGISTERED EVENTS** in the **Data360 > Events** section in VWO. To know more about Events in VWO Data360, [click here](https://help.vwo.com/hc/en-us/articles/8676443712537-Working-with-Events-in-VWO){:target="_blank"}.

## Using VWO Cloud mode destination with VWO FullStack

To use the VWO Cloud mode destination with the VWO FullStack suite, you first need to link your VWO FullStack environment with Segment using the environment’s SDK key. Post this, you have to integrate your VWO account with Segment.

To link your VWO FullStack environment with Segment, perform the following steps:

1. From your VWO dashboard, navigate to the navbar on the left > **FullStack > Projects** and select the appropriate project.

> 2. Under the **Environments** section, click the **Copy** button corresponding to the environment that you want to link to Segment.

Author: The copy button is for the SDK key which corresponds to the respective environments.

1. Initialize VWO FullStack SDK. Follow the steps mentioned [here](https://developers.vwo.com/docs/quick-start-guide){:target="_blank"}, as per your server application.
2. In order to track your visitors in VWO, you need to provide the user IDs of the visitors, which were used to track them in the VWO FullStack campaign. You need to pass that same User ID as `vwo_uuid` with all the calls. The `vwo_uuid` key must be included in *properties* for Track and Page API calls whereas it should be added in *traits* for Identify API calls. 
3. All the events triggered in Segment will be available under **UNREGISTERED EVENTS** section which can be accessed by navigating from the left navbar > **Data360 > Events**.

## Supported Segment Calls in VWO

VWO Supports the following calls, as specified in the [Segment Spec](https://segment.com/docs/connections/spec/){:target="_blank"}.

### Track
The [Track](https://segment.com/docs/connections/spec/track/){:target="_blank"} API call is used to record any actions your visitors perform, along with any properties that describe the action. Each action is known as an event. 

> The destination forwards these events to VWO Data360 where they can be seen under the **UNREGISTERED EVENTS** section in **Data360 > Events** in VWO. These events can be registered and used as [Metrics](https://help.vwo.com/hc/en-us/articles/8675547113625-Working-with-Metrics-in-VWO){:target="_blank"} in VWO.

Author: The events are not registered by default. The user can register them if they want to use the events in VWO otherwise they can be left unregistered.

**Sample payload for Track Call to the destination**

```js
analytics.track("Segment Test Event", {
	"vwo_uuid": "<VWO UUID or User ID>",
	"property1": "value1"
});
```

**Corresponding JavaScript event that would generate the above payload**

```js
{
  "type": "track",
  "event": "<Event Name>",
  "properties": {
    "vwo_uuid": "<VWO UUID or User ID>",
    "property1": "value1"
  }
}
```

> info ""

> > The event name will be prepended with “**segment.**” before sending it to VWO. So if an event named "**ctaClick**" is triggered in Segment, it’ll appear as "**segment.ctaClick**" under **UNREGISTERED EVENTS** in VWO.

Reviewer: ```suggestion
> Event names are prepended with `segment.` before they're sent to VWO. If an event named "**ctaClick**" is triggered in Segment, it appears as `segment.ctaClick` under **UNREGISTERED EVENTS** in VWO.
```

### Identify

> The [Identify](https://segment.com/docs/connections/spec/identify/){:target="_blank"} call lets you associate a visitor with their actions and capture their traits. 

Reviewer: ```suggestion
The [Identify](/docs/connections/spec/identify/) call associates a visitor with their actions and captures their traits. 

```

> The destination forwards these traits to VWO Data360 where they can be seen under the **UNREGISTERED ATTRIBUTES** section in the **Data360 > Attributes** in VWO. These attributes can be registered and used to create [segments in VWO](https://help.vwo.com/hc/en-us/articles/8976459309465-Working-with-Segments-in-VWO-Data360){:target="_blank"}. To know more about Attributes in VWO Data360, [click here](https://help.vwo.com/hc/en-us/articles/8681465703705-Working-with-Attributes-in-VWO){:target="_blank"}.

Reviewer: ```suggestion
The destination forwards these traits to VWO Data360 where they can be seen under the **UNREGISTERED ATTRIBUTES** section in the **Data360 > Attributes** in VWO. These attributes can be registered and used to create [segments in VWO](https://help.vwo.com/hc/en-us/articles/8976459309465-Working-with-Segments-in-VWO-Data360){:target="_blank"}. For more information about Attributes in VWO Data360, see [Working with Attributes in VWO](https://help.vwo.com/hc/en-us/articles/8681465703705-Working-with-Attributes-in-VWO){:target="_blank"}.
```

**Sample payload for Identify call to the destination**

```js
{
  "type": "identify",
  "traits": {
    "vwo_uuid": "<VWO UUID or User ID>",
    "trait1": "value1"
  }
}
```

**Corresponding JavaScript event that would generate the above payload**

```js
analytics.identify({
	"vwo_uuid": "<VWO UUID or User ID>",
	"trait1": "value1"
});
```

> info ""

> > Each trait key will be prepended with “**segment.**” before sending to VWO. So if a trait named "**trait1**" is sent in Segment, it’ll appear as "**segment.trait1**" under UNREGISTERED ATTRIBUTES in VWO.

Reviewer: This is mentioned above. Recommend deleting here.

Author: The message shown above is specific to the event and event names. This is for the traits and how they'll be displayed in VWO's attributes.

### Page

> The [Page](https://segment.com/docs/connections/spec/page/){:target="_blank"} call lets you record whenever a visitor arrives at a page of your website, along with any optional properties about the page. On receiving this call, the destination triggers VWO’s Page Visit event.

Reviewer: ```suggestion
The [Page](/docs/connections/spec/page/){:target="_blank"} call records when a visitor arrives at a page of your website, along with any optional properties about the page. When received, the destination triggers VWO’s Page Visit event.
```

> info ""

> > It should only be used when working with web pages and not when using server-side sources as Page Visit event is not supported in VWO's FullStack Suite.

Reviewer: ```suggestion
> Use Page calls with web pages only. Server-side sources in VWO's FullStack Suite do not support the Page Visit event.
```
